c_emulator/riscv_sim.c

```diff
@@ -1088,15 +1093,29 @@ int main(int argc, char **argv)
     }
     printf("Connected\n");
   } else
-    entry = load_sail(file);
+    entry = load_sail(initial_elf_file);
 #else
-  uint64_t entry = load_sail(file);
+  uint64_t entry = load_sail(initial_elf_file);
 #endif
+  /* Load any additional ELF files into memory */
+  for (int i = files_start + 1; i < argc; i++) {
+    bool is32bit;
+    uint64_t additional_entry;
+    fprintf(stdout, "Loading additional ELF file %s.\n", argv[i]);
+    /*
+     * We use load_elf() directly instead of load_sail() as we do not want to
+     * override tohost/{begin,end}_signature from the initial ELF file.
+     */
+    load_elf(argv[i], &is32bit, &additional_entry);
+    check_elf(is32bit);
+    fprintf(stdout, "Additional ELF %d Entry @ 0x%" PRIx64 "\n",
+            i - files_start, additional_entry);
```

Comment:
This isn't meaningful to print, the Sail model isn't doing anything with it, and likely neither are the other ELF files

Reply:
Fair enough - I thought it could be helpful to print the S-mode entry point, but I've removed it now. You can always use `readelf` to get that info...

Reply:
For that specific use case it'll be virtual rather than physical anyway, which isn't much help until you're already there